Compile problem with ruby preview 2.4.0 - possibly related to the upcoming Bignum/Fixnum change/unification

Hello Aaron and others,

I have tried latest ruby 2.4.0-preview1. It may be not worth the
time to invest for previews - however had, perhaps even in the
final 2.4.0 release, this issue may persist, hence why I report
it.

Command (works perfectly fine on ruby 2.3.1 by the way):

> gem install syck-1.1.0.gem
 
**Building native extensions.  This could take a while...**

ERROR:  Error installing syck-1.1.0.gem:
        ERROR: Failed to build gem native extension.

current directory: /Programs/Ruby/2.4.0preview1/lib/ruby/gems/2.4.0/gems/syck-1.1.0/ext/syck
/Programs/Ruby/2.4.0preview1/bin/ruby -r ./siteconf20160624-21263-8tem4.rb extconf.rb

checking for st.h... yes
creating Makefile

To see why this extension failed to compile, please check the mkmf.log which can be found here:

/Programs/Ruby/2.4.0preview1/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0/syck-1.1.0/mkmf.log

current directory: /Programs/Ruby/2.4.0preview1/lib/ruby/gems/2.4.0/gems/syck-1.1.0/ext/syck
make "DESTDIR=" clean

current directory: /Programs/Ruby/2.4.0preview1/lib/ruby/gems/2.4.0/gems/syck-1.1.0/ext/syck
make "DESTDIR="
compiling bytecode.c
compiling emitter.c
compiling gram.c
compiling handler.c
compiling implicit.c
compiling node.c
compiling rubyext.c

**The error appears to be:**

    rubyext.c: In function 'syck_resolver_transfer':
    rubyext.c:1165:34: error: 'rb_cBignum' undeclared (first use in this function)
                     if ( subclass == rb_cBignum )
                                  ^
    rubyext.c:1165:34: note: each undeclared identifier is reported only once for 
    each function it appears in make: *** [Makefile:241: rubyext.o] Error 1

    make failed, exit code 2

Gem files will remain installed in /Programs/Ruby/2.4.0preview1/lib/ruby/gems/2.4.0/gems/syck-1.1.0 for inspection.
Results logged to /Programs/Ruby/2.4.0preview1/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0/syck-1.1.0/gem_make.out

I am not sure why this error happens, perhaps because of the Bignum/Fixnum change? My C is
very weak but perhaps there may have to be a check added to see whether rb_cBignum is
available or not. But I am just guessing here really, anyway.

At any rate, this is just a small report, feel free to close it at any time.

Thanks!

--- a/ext/syck/rubyext.c
+++ b/ext/syck/rubyext.c
@@ -1162,7 +1162,11 @@ syck_resolver_transfer(VALUE self, VALUE type, VALUE val)
             }
             else if ( !NIL_P( target_class ) )
             {
+#ifdef RUBY_INTEGER_UNIFICATION
+                if ( subclass == rb_cInteger )
+#else
                 if ( subclass == rb_cBignum )
+#endif
                 {
                     obj = rb_str2inum( val, 10 ); /* for yaml dumped by 1.8.3 [ruby-core:6159] */
                 }
@@ -2325,4 +2329,3 @@ Init_syck()
     rb_define_method( cEmitter, "set_resolver", syck_emitter_set_resolver, 1);
     rb_define_method( cEmitter, "node_export", syck_emitter_node_export, 1);
 }
-

